source/common/stream_info/filter_state_impl.cc

```diff
@@ -31,5 +31,32 @@ const FilterState::Object* FilterStateImpl::getDataGeneric(absl::string_view dat
   return it->second.get();
 }
 
+void FilterStateImpl::addToListGeneric(absl::string_view list_name,
+                                       std::unique_ptr<Object>&& item) {
+  // TODO(Google): Remove string conversion when fixed internally.
```

Comment:
Is this the right comment? I think this is the string conversion issue in public abseil, not the proto string mess? Same below. TBH not sure it's worth commenting the abseil string conversion issue but up to you.

Reply:
err, I don't even know what the issue is :). I merely copied the comments on top of the std::string(data_name) when I added similar code. https://github.com/envoyproxy/envoy/pull/4588/files#diff-d0929a26ef8f3accf4b599d02387709cR20

Should I remove all these comments?

Reply:
There are 2 different issues here:
1) Google has some internal string class that their internal protobuf uses. This has caused as a bunch of issues with having to make copies where we otherwise would not.
2) absl::string_view does not auto cast to std::string. That's this issue.

Yes I would just remove these comments.

Reply:
My memory is that there's a way to solve this cleanly with C++17, but we can't do this internally.  Please leave a TODO(Google) so that we can clean this up when we upgrade our C++ environment.

Reply:
FYI we are only on C++14 for Envoy, not sure when we will go to 17. I'm fine with leaving the comments if you want, but I would clarify which issue we are talking about (either 1 or 2).

Reply:
So I dug up my notes.  The problem is that internal to Google we aren't using C++14, instead we're in the (long) process of jumping to C++17.  The functionality that envoy has that we don't internally is templated arguments to find() to allow use of an absl:string_view.  (See (3) and (4) entries under https://en.cppreference.com/w/cpp/container/map/find).  So it's neither (1) or (2), but it's own issue. 

Suggestion: "TODO(Google): Remove string conversion when templated map::find() is available internally".

Reply:
Ah. Thanks for the clarification. OK makes sense to me. :)

Reply:
I simply copied the comments from existing code, which I remember was added by you (@curiouserrandy). so the question is, why change the comment now? :).

Reply:
Because @mattklein123 complained about the old one, of course!  :-}

I have no objection to leaving the comment the way it is, or if Matt wants it changed, to doing so in a separate PR (which I'll write).  Up to you/him.

Reply:
It's not a big deal to leave it. It's mostly that there are multiple Google string issues and I'm having a hard time keeping track!